### PR TITLE
Add `deeparg`

### LINKED ISSUE #1186: new module: deeparg
I think it would be good to have a module for [deepARG](https://bioconda.github.io/recipes/deeparg/README.html)

- [x] This module does not exist yet with the [`nf-core modules list`](https://github.com/nf-core/tools#list-modules) command
- [x] There is no [open pull request](https://github.com/nf-core/modules/pulls) for this module
- [x] There is no [open issue](https://github.com/nf-core/modules/issues) for this module
- [x] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by jfy133**
## PR checklist

Closes #1186

Adds both download data(base) and predict. The former is required for the tests for latter.

Note: download data takes a very long time, so CI tests will be slow.

- [x] This comment contains a description of changes (with reason).
- [ x If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: modules/deeparg/predict/meta.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      pattern: "*.align.daa"
+  - daa_tsv:
+      type: file
+      description: Alignments scores against ARG-like sequences from DIAMOND alignment
+      pattern: "*.align.daa.tsv"
+  - arg:
+      type: file
+      description: Table containing sequences with an ARG-like probability of more than specified thresholds
+      pattern: "*.mapping.ARG"
+  - arg:
```
Comment (jfy133) [inline modules/deeparg/predict/meta.yml:62]:
```suggestion
  - potential_arg:
```

**Review by maxulysse — APPROVED** [review]
:+1:

File: modules/deeparg/predict/main.nf [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::deeparg=1.0.2" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity//deeparg:1.0.2--pyhdfd78af_1' :
+        'quay.io/biocontainers/deeparg:1.0.2--pyhdfd78af_1' }"
+
+    input:
+    tuple val(meta), path(fasta), val(model)
+    tuple path(db)
```
Comment (mashehu) [inline modules/deeparg/predict/main.nf:14]: not sure if the tuples here are correct?

**Review by mashehu — APPROVED** [review]

File: modules/deeparg/predict/main.nf (hunk 2 above)
Comment (jfy133) [reply modules/deeparg/predict/main.nf:14]: Oh no, why?

Comment (mashehu) [reply modules/deeparg/predict/main.nf:14]: A tuple with just one element (`path(db)`) seems to be a bit of an overkill, no?

Comment (jfy133) [reply modules/deeparg/predict/main.nf:14]: Ah crap yeah! I'll fix that tomorrow

Comment (jfy133) [reply modules/deeparg/predict/main.nf:14]: You can have that review ;)

**Pull request merged**